src/stage1/ir.cpp

```diff
@@ -9820,6 +9820,38 @@ static ErrorMsg *ir_eval_math_op_scalar(IrAnalyze *ira, Scope *scope, AstNode *s
                 float_min(out_val, op1_val, op2_val);
             }
             break;
+        case IrBinOpSatAdd:
+            if (is_int) {
+                bigint_add_sat(&out_val->data.x_bigint, &op1_val->data.x_bigint, &op2_val->data.x_bigint, type_entry->data.integral.bit_count, type_entry->data.integral.is_signed);
+            } else {
+                // TODO: support floats?
+                zig_panic("float types not supported by @addWithSaturation");
```

Comment:
Can you make this a compile error rather than a panic? (test coverage for the compile error too please!)

Reply:
how can i add test coverage for the compile error? i remember that this is a thing but can't seem to find an example.

Reply:
Add a case to https://github.com/ziglang/zig/blob/master/test/compile_errors.zig

You can run them with `./zig build test-stage2`

Reply:
Now that i've added the compile error test coverage, the actual error reported is `"error: invalid operands to binary expression: 'f32' and 'f32'"`.  Should this actually be changed to `zig_unreachable()`?

Reply:
Yep!